### ISSUE TITLE
FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory

### ISSUE THREAD

**Issue by SmirnovW** [matmuchrapna/gulp-bem-stub#21]
When i run "Gulp build", i get "FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory".

**Comment by iamstarkov** [comment]
can you tell us your environment `node -v` and `npm -v` and OS name with version. thanks

**Comment by SmirnovW** [comment]
It was necessary to say at once, sorry...
npm: 2.14.2
node: v4.0.0
os: windows 8.1
Now i see i have node v4.0.0. but now exist 4.1.1, maybe problem in this?
Thanks!

**Comment by iamstarkov** [comment]
maybe, try it. keep us informed

**Comment by SmirnovW** [comment]
Nope, still error "FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
".

**Comment by DelphiAnton** [comment]
Я нашел проблему. 
gulp-project\node_modules\gulp-bem\node_modules\stream-array\index.js

Конкретно в методе:
StreamArray.prototype._read = function(size) {
    this.push(this._queue.shift());
};

Там массив не очищался, а undefindef забивался. Пока опыта мне грамотно написать не хватило, я просто заменил на этот, модуль из корня:
gulp-project\node_modules\stream-array\index.js

**Comment by lilliputten** [comment]
@DelphiAnton Thanks! Had same problem with old `stream-array` (0.1.3). Solved by replacing by a new package version.

**Comment by nikolai-shabalin** [comment]
Видимо, всё правильно пишет народ, что проблема в `stream-array`.

Моё окружение 
1) node v6.2.2
2) npm 3.9.5

Как я решил
1) клоним проект
2) npm i
3) удаляем папку пакета `node_modules/stream-array`
4) Начинаем править файл  `node_modules/gulp-bem/package.json`
`на 58 строке "stream-array": "^0.1.3", меняем на  "stream-array": "1.1.2",`
5) и опять делаем `npm i`
6) `gulp watch`

**Comment by SmirnovW** [comment]
@nikolai-shabalin and everybody who has problem with BEM tools for gulp, this is for you guys! After a huge pain with gulp-bem-stub and other tools for BEM, we in our company created our own tool for BEM. This is open sourse version: https://github.com/pbelyaev/extract-bem
